js/create_data.js

```diff
@@ -91,62 +118,32 @@ function roundToTwo(num) {
     return +(Math.round(num + "e+2") + "e-2");
 }
 
+// ****
+// MJN: using indexes is fine with hardcoded data,
+// but in general with arrays you want to "push" and "pop"
+// values from them.
+function addRent(tenant, rent) {
+  tenant.rents.push(rent);
+}
+// ****
+
 // builds data for two tenants, as per the specification
 // Tenant 1 hard coded data
 {
-    var tenants = [];
-    tenants[0] = new Tenant();
-    tenants[0].name = "Furniture";
-    tenants[0].unit_size = 20000;
-    tenants[0].rents = [];
+  var tenants = [];
 
-    //rent year 1
-    tenants[0].rents[0] = new Rent();
-    tenants[0].rents[0].startDate = new Date("04/01/2016");
-    tenants[0].rents[0].endDate = new Date("03/31/2017");
-    tenants[0].rents[0].monthlyRent = 44166;
+  var baseRent = 44166;
 
-    // rent year 2
-    //add roundings........
-    tenants[0].rents[1] = new Rent();
-    tenants[0].rents[1].startDate = new Date("04/01/2017");
-    tenants[0].rents[1].endDate = new Date("03/31/2018");
-    tenants[0].rents[1].monthlyRent = roundToTwo(tenants[0].rents[0].monthlyRent * 1.025);
+  var tenant1 = new Tenant("Furniture", 20000, []);
 
-    // rent year 3
-    //add rounding..............
-    tenants[0].rents[2] = new Rent();
-    tenants[0].rents[2].startDate = new Date("04/01/2018");
-    tenants[0].rents[2].endDate = new Date("03/31/2019");
-    tenants[0].rents[2].monthlyRent = roundToTwo(tenants[0].rents[1].monthlyRent * 1.025);
+  for (var i = 0; i < 6; i++) {
```

Comment:
Here you can see our `addRent` function at work; we no longer push values directly into a specific index of the array (which is very, very easy to get wrong) but we "push" new rent values onto the end of the array.